src/apis/stockx/index.js

```diff
@@ -0,0 +1,205 @@
+
+import { jar } from 'request-promise-native';
+
+// APIs
+import ProductsApi from './products';
+import AsksApi from './asks';
+import BidsApi from './bids';
+import UserApi from './user';
+
+// Utils
+import { format, currencies } from '../../utils';
+
+export default class StockX {
+
+  /**
+   *
+   * @param {Object} options
+   * @param options.proxy - The proxy to use for future requests
+   * @param options.currency - The currency used when making requests
+   */
+  constructor(options = {}){
+    const { proxy = null, currency = currencies.USD } = options;
+
+    //Configurable options
+    this.currency = currency;
+    this.proxy = proxy ? format(proxy) : null;
+    
+    this._bearer = null;
+    this._isLoggedIn = false;
+
+    this._context = {
+      currency: this.currency,
+      jar: jar(),
+      headers: {
+        accept: '*/*',
+        'accept-language': 'en-US,en;q=0.9',
+        appos: 'web',
+        appversion: '0.1',
+        authority: 'stockx.com',
+        host: 'stockx.com',
+        origin: 'https://stockx.com',
+        'sec-fetch-mode': 'cors',
+        'sec-fetch-site': 'same-origin',
+        'upgrade-insecure-requests': 1,
+        'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36',
+        'x-requested-with': 'XMLHttpRequest',
+      },
+      proxy: this.proxy,
+      bearer: this._bearer,
+      isLoggedIn: this._isLoggedIn,
```

Comment:
Why maintain these on the two instances of `_context` and `this` and not just `_context`?

Reply:
Good catch! This was before the shared data singleton, so I'll make those changes now.